test/extended/build_context_dir_test.go

```diff
@@ -0,0 +1,76 @@
+// +build integration,extended,!no-etcd,!no-docker,docker
```

Comment:
I think we are overusing those, especially in that we do copy&paste for other integration tests. I think it would make sense to clear this out doing this. Since for this test you are requiring etcd and docker this line should state:
```
// +build integration,extended,etcd,docker
```
IMHO